addons/api/addon/adapters/application.js

```diff
@@ -118,12 +118,41 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
 
   /**
    * Intercepts "empty" responses and adds an empty `items` array.
+   * This query method now supports pagination by checking the responseType and
+   * passing in the refresh token if the responseType indicates more data
    * @override
    * @method query
    * @return {Promise} promise
    */
-  query() {
-    return super.query(...arguments).then(prenormalizeArrayResponse);
+  async query(store, schema, query) {
+    let result;
```

Comment:
Should this start as { items: [] }? And we can push to the array rather than using a data field? Do we need it separated for any reason?

Reply:
I separated to avoid adding duplicates to the results items array. Otherwise, I had to do another check to ensure I wasn't adding duplicated results to the item array, which seemed more readable.

Reply:
What do you mean by duplicates in this case?

Reply:
After a chat with @DhariniJeeva over zoom I understand this issue better and I agree we will need 2 data structures. One `result` to keep the entire result of the query and `data` to keep building an array of `result.items` per iteration.

Reply:
Since we are reassigning the `result`, if we don't have another data structure to store the total result, we will only be overwriting the previous entry in the result, and pushing items to that result can lead to duplicates as a side effect